qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/domain/TestActivityDescription.kt

```diff
@@ -3,5 +3,7 @@ package com.mapbox.navigation.qa_test_app.domain
 data class TestActivityDescription(
     val label: String,
     val fullDescriptionResource: Int,
-    val launchActivityFun: LaunchActivityFun
+    /** Set to true if the example expects permissions granted before launching. */
+    val handlesPermissions: Boolean = false,
```

Comment:
The wording of the comment and the flag is confusing to me.  Rather than using a flag I suggest creating an interface for the special case activities to implement that way the other activities don't have to know anything about it and in the future we don't have to think about this feature unless it applies to an activity we're adding.  The permission checking code can check if a given activity is a `specialCaseActivity` and act accordingly.

Reply:
I'm also a little confused on the recommendation here, is it to have a different mechanism to represent the `TestActivityDescription`? Or is it to create a different `LaunchActivityFun`?

Reply:
I shouldn't have put both comments together.  

1. I found the current wording of the comment and the flag confusing.  If `handlesPermissions` is true wouldn't that mean the activity handles the permission checking/requesting?  That's how it reads to me but if that's true then it seems to contradict the comment above that the activity expects the permissions to already be in place.

2. I think a better alternative to the flag is to create an "IamSpecial" interface for the activities that need this permissions handling you've implemented.  Using the flag means every time we add a new activity to the app. we have to remember what that flag was for and in 6 months we're going to forget.  Instead of checking for the flag check the type of the activity to see if it implements the "special permissions" interface.  

This code here:

```
if (element.handlesPermissions)  {  // This changes to is activity special ?
                element.launchActivityFun(this)
            } else {
                launchAfterPermissionResult(element)
            }

```

```
interface SpecialPermissions (give it a better name)

SomeTestActivity : SpecialPermissions {
...
} 
```

This also eliminates the first issue I raised.  Just my 2 cents.

Reply:
Oh right, I was trying a few ideas to make it clear what `handlesPermissions` does. But looks like it's still confusing. I'm updating it to see if it makes more sense. Let's see if that works.

Interesting idea on adding an interface, seems like the interface may want to specify the `launchActivityFun` 🤔 . I thought it was expected to have the condition live close to the `TestActivityDescription.launchActivityFun`. Obviously it works, but arguable that it is preferred? Do you have a strong opinion on this approach?

Reply:
No strong opinion I just prefer not to have to think about this special case unless it is a special case rather than always having to think about it.  But, it's personal preference.